### PR TITLE
Origin/error msg

### PR DESCRIPTION
**Pull request by bwan2**
dependency [PR11](https://github.com/pinocchio61/pinocchioNodeServer/pull/11)

Standardized error messages and responses. Error messages will always have the following format: 
```
{
    "success": false,
    "errors": [
        {
            "msg": "Error message here",
            "param": "what param the error happened on"
        }
    ]
}
```

### CODE REVIEW

File: src/utils/errorMsg.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// parameters
```
Comment (miajhaochen) [inline src/utils/errorMsg.js:1]: you can separate them to 
```js
exports.params = {
  EMAIL: 'email',
  PASSWORD: 'password',
  FIRSTNAME: 'firstname',
  ...
};

exports.messages. = {
  EMAIL_REGISTERED: 'The email provided is already registered.',
  EMAIL_NOT_REGISTERED: 'The email provided is not registered.'
  ... 
};
```

In this way, when using these constants, we can do it like `errorMsg.params.EMAIL` or `errorMsg.messages.EMAIL_REGISTERED`, which is clearer than `errorMsg.EMAIL` or `errorMsg.EMAIL_REGISTERED`. 

Of course we can have a consensus that params will be named with only a single word, which means variables with underscores are messages. However, we cannot make sure whether a single word will be enough for all the params variables.

File: src/controllers/accounts.controller.js [hunk 2]
```diff
@@ -43,7 +45,7 @@ exports.register = async (req, res, next) => {
     const validation = validationResult(req);
     if (!validation.isEmpty()) {
       const msgs = validation.errors.map(err => `The ${err.param} has incorrect format.`)
-      return res.status(422).json({success: false, errors: msgs});
+      return res.status(422).json(responseMsg.validationError422(validation.errors));
```
Comment (miajhaochen) [inline src/controllers/accounts.controller.js]: line 47 can be removed

File: src/controllers/events.controller.js [hunk 3]
```diff
@@ -43,11 +46,11 @@ exports.create = async (req, res, next) => {
     const validation = validationResult(req);
     if (!validation.isEmpty()) {
       const msgs = validation.errors.map(err => `The ${err.param} has incorrect format.`)
-      return res.status(422).json({success: false, errors: msgs});
+      return res.status(422).json(responseMsg.validationError422(validation.errors));
```
Comment (miajhaochen) [inline src/controllers/events.controller.js]: line 48 can be removed

File: src/controllers/accounts.controller.js [hunk 4]
```diff
@@ -87,59 +89,59 @@ exports.register = async (req, res, next) => {
...
 
 // log into an account
 exports.login = async (req, res, next) => {
     // check whether inputs are valid
     const validation = validationResult(req);
     if (!validation.isEmpty()) {
-      const msgs = validation.errors.map(err => `The ${err.param} has incorrect format.`)
-      return res.status(422).json({success: false, errors: msgs});
+      const msgs = validation.errors.map(err => `The ${err.param} has incorrect format.`);
+      return res.status(422).json(responseMsg.validationError422(validation.errors));
```
Comment (miajhaochen) [inline src/controllers/accounts.controller.js]: line 119 can be removed

**Comment by miajhaochen** [comment]
conflict with master branch!

**Comment by miajhaochen** [comment]
conflict ~~
 please merge with master branch~~

File: src/controllers/accounts.controller.js (hunk 4 above)
Comment (bwan2) [reply src/controllers/accounts.controller.js]: removed

File: src/controllers/events.controller.js (hunk 3 above)
Comment (bwan2) [reply src/controllers/events.controller.js]: removed

File: src/controllers/accounts.controller.js (hunk 2 above)
Comment (bwan2) [reply src/controllers/accounts.controller.js]: removed

File: src/utils/errorMsg.js (hunk 1 above)
Comment (bwan2) [reply src/utils/errorMsg.js:1]: changed

File: src/controllers/accounts.controller.js [hunk 5]
```diff
@@ -42,8 +44,7 @@ exports.register = async (req, res, next) => {
   // check whether inputs are valid
   const validation = validationResult(req);
   if (!validation.isEmpty()) {
-    const msgs = validation.errors.map((err) => `The ${err.param} has incorrect format.`);
-    return res.status(422).json({success: false, errors: msgs});
+    return res.status(422).json(responseMsg.validationError422(validation.errors));
```
Comment (miajhaochen) [inline src/controllers/accounts.controller.js:47]: the msgs returned by `express-validator` are only `Invalid value`. I cannot print it on the screen directly looool
(So originally, I modified it to `The ${err.param} has incorrect format.` at line 45)
maybe map the error message in function `validationError422 ` ? 

image

Comment (bwan2) [reply src/controllers/accounts.controller.js:47]: This is fixed in #16

**Pull request merged**